### ISSUE TITLE
Add `TAU` to `approx_constant` lint

### ISSUE THREAD

**Issue by mikerite** [rust-lang/rust-clippy#7641]
Lint name: approx_constant

I tried this code:

```rust
let two_pi = 6.2833;
```

I expected to see this happen:
`approx_const` should recommend using `TAU` if my version setting is ≥ 1.47.

Instead, this happened:
Nothing. It's a false negative.

**Comment by mikerite** [comment]
Fixed by #7642
